From 9ee78b29857be9f01ae0ae36b145136192033ba2 Mon Sep 17 00:00:00 2001
From: Patrick Griffis <tingping@tingping.se>
Date: Sun, 24 Dec 2017 12:27:06 -0500
Subject: [PATCH] appdata: Provide app-id

---
 data/misc/hexchat.appdata.xml.in | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/data/misc/hexchat.appdata.xml.in b/data/misc/hexchat.appdata.xml.in
index 8b5b3ee8..777bb58b 100644
--- a/data/misc/hexchat.appdata.xml.in
+++ b/data/misc/hexchat.appdata.xml.in
@@ -1,6 +1,9 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <component type="desktop">
   <id>hexchat.desktop</id>
+  <provides>
+    <id>hexchat.desktop</id>
+  </provides>
   <name>HexChat</name>
   <developer_name>HexChat</developer_name>
   <metadata_license>CC0-1.0</metadata_license>
-- 
2.14.3

